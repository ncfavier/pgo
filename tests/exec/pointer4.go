package main
import "fmt"

func foo() *int { x := 41; return &x }

func main() {
	x := foo()
	y := foo()
	fmt.Print(*x, "\n")
	fmt.Print(*y, "\n")
	*x = 42
	*y = 43
	fmt.Print(*x, "\n")
	fmt.Print(*y, "\n")
}
