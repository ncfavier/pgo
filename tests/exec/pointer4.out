41
41
42
43
